### PR TITLE
ddl: fast fail check for create index to accelerate CI

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Make the CI run faster !

```
➜  session git:(master) ✗ GO111MODULE=on go test  -check.f TestIndexMaxLength
before:
ok  	github.com/pingcap/tidb/session	18.229s
after:
ok  	github.com/pingcap/tidb/session	0.146s
```

### What is changed and how it works?

If DDL check fail before the job is put to job queue, the fail path logic is super fast.
After DDL job is put to the queue, and if the check fail, TiDB will run the DDL cancel process, 
and the recover step cause DDL wait few seconds, that's painfully slow.

In unit test session.TestIndexMaxLength, it cover a index key lengh too long case, and triggers the DDL fail logic.
Each time that logic is triggered, it waits about 3s ... the total test runs 18s

This commit add a check for create index before the job is put to the job queue.

 - No code

@winkyao @zimulala @jackysp

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8592)

### CODE REVIEW

**Comment by tiancaiamao** [comment]
B.T.W, this change makes the session package test time reduce to 16.417s, from 53.792s

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -2035,6 +2035,12 @@ func (d *ddl) CreateIndex(ctx sessionctx.Context, ti ast.Ident, unique bool, ind
 		return errors.Trace(err)
 	}
 
+	// Check before put the job is put to the queue.
+	_, err = buildIndexColumns(t.Meta().Columns, idxColNames)
```
Comment (winkyao) [inline ddl/ddl_api.go]: Please add more comments, explain why we do a check here.

Comment (tiancaiamao) [reply ddl/ddl_api.go]: Done

**Pull request merged**